### PR TITLE
sci-astronomy/xephem: EAPI7, use HTTPS, add missing eclass, fix SRC_URI

### PR DESCRIPTION
**Pull request by mm1ke**
Hi,

This updates the ebuild to use https and adds the missing `desktop` eclass.
Since the SRC_URI wasn't working anymore I've also added a EAPI7 version (with some cleanup) and updated the SRC_URI. I haven't updated the orginial ebuild since the new source differ by 2 byte's but other than that it's the same source. (and even is provided by the original homepage)

```diff
4c4
< EAPI=6
---
> EAPI=7
6c6
< inherit desktop eutils toolchain-funcs
---
> inherit desktop toolchain-funcs
10c10
< SRC_URI="http://97.74.56.125/free/${P}.tar.gz"
---
> SRC_URI="https://www.clearskyinstitute.com/xephem/${P}.tgz"
14,15c14
< KEYWORDS="amd64 ppc ppc64 x86 ~amd64-linux ~x86-linux"
< IUSE=""
---
> KEYWORDS="~amd64 ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux"
22a22,24
> HTML_DOCS=( GUI/xephem/help/. )
> DOCS=( README )
> 
31,32c33,34
<       echo > "${T}"/XEphem "XEphem.ShareDir: /usr/share/${PN}"
<       echo > "${T}"/99xephem "XEHELPURL=/usr/share/doc/${PF}/html/xephem.html"
---
>       echo > "${T}"/XEphem "XEphem.ShareDir: /usr/share/${PN}" || die
>       echo > "${T}"/99xephem "XEHELPURL=/usr/share/doc/${PF}/html/xephem.html" || die
42d43
<       has_version '<x11-base/xorg-x11-7.0' && insinto /etc/X11/app-defaults
45c46
<       dodoc README
---
>       einstalldocs
53,54d53
<       insinto /usr/share/doc/${PF}/html
<       doins -r help/*
```

### CODE REVIEW

File: sci-astronomy/xephem/xephem-3.7.7-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+DESCRIPTION="Interactive tool for astronomical ephemeris and sky simulation"
+HOMEPAGE="https://www.clearskyinstitute.com/xephem"
+SRC_URI="https://www.clearskyinstitute.com/xephem/${P}.tgz"
+
+LICENSE="XEphem"
+SLOT=0
+KEYWORDS="~amd64 ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux"
+
+RDEPEND=">=x11-libs/motif-2.3:0
+	virtual/jpeg:0=
```
Comment (SoapGentoo) [inline sci-astronomy/xephem/xephem-3.7.7-r1.ebuild:17]: `:0` only (this is a virtual)

File: sci-astronomy/xephem/xephem-3.7.7-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	media-libs/libpng:0="
+DEPEND="${RDEPEND}
+	sys-apps/groff"
```
Comment (SoapGentoo) [inline sci-astronomy/xephem/xephem-3.7.7-r1.ebuild:20]: `groff` needs to go into `BDEPEND`

File: sci-astronomy/xephem/xephem-3.7.7-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+src_compile() {
+	tc-export CC AR RANLIB
+	emake -C GUI/xephem
+}
+
+src_install() {
+	insinto /usr/share/X11/app-defaults
+	doins "${T}"/XEphem
+	doenvd "${T}"/99xephem
```
Comment (SoapGentoo) [inline sci-astronomy/xephem/xephem-3.7.7-r1.ebuild:45]: consider ditching both these and doing the `newins -` `newenvd -` trick without resorting to temporaries

File: sci-astronomy/xephem/xephem-3.7.7-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	einstalldocs
+
+	cd GUI/xephem
```
Comment (SoapGentoo) [inline sci-astronomy/xephem/xephem-3.7.7-r1.ebuild:48]: missing `|| die`

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

**Comment by mm1ke** [comment]
@SoapGentoo 
Updated with all your suggestions/fixes included :)

**Comment by mm1ke** [comment]
ping
any chance we can merge this? :)

**Comment by ConiKost** [comment]
Could you update to solve merge conflict?

**Comment by mm1ke** [comment]
@ConiKost done, should be fine now to merge :)

**Review by ConiKost — APPROVED** [review]

**Pull request closed without merging**